addon/index.js

```diff
@@ -32,6 +32,26 @@ function maybeUnwrapProxy(o) {
   return isProxy(o) ? maybeUnwrapProxy(safeGet(o, 'content')) : o;
 }
 
+function tryContent(ctx) {
+  return ctx._content ? ctx._content : ctx.content ? ctx.content : ctx;
+}
+
+function deepNotifyPropertyChange(obj, path) {
+  var paths = path.split('.');
+  var lastPath = paths.pop(),
+    current = obj,
+    i;
+
+  for (i = 0; i < paths.length; ++i) {
+    if (current[paths[i]] != undefined) {
+      current = current[paths[i]];
+    } else {
+      notifyPropertyChange(tryContent(current), paths[i]);
+    }
```

Comment:
Rewrote my last comment as my understanding changed.

The idea is to traverse down to find the deepest `defined` ancestor. 

example 1.
Take: key as `foo.bar.baz` and data as `{ foo: { bar: {} } }`

The output should: `notifyPropertyChange(foo.bar, 'baz')`

example 2.
Take: key as `foo.bar.baz` and data as `{ }`, everything `undefined`

The output should: `notifyPropertyChange(data, 'foo')`

I think this could be optimized like this:
```ts
for (i = 0; i < paths.length; ++i) {
    if (current[paths[i]] != undefined) { // we can still go deep
      current = current[paths[i]];
    } else {
      //we just found that its a dynamic set, so we notify here and stop, no point in going further. 
      notifyPropertyChange(tryContent(current), paths[i]);
      return;
    }
  //We didn't stop, we haven't notified, and we went the deepest as possible, so we notify there.
  notifyPropertyChange(tryContent(current), lastPath);
}
```
I think its ok to start from the top, because we can early notify and return